### ISSUE TITLE
Search Layers and Multilayers

### ISSUE THREAD

**Issue by BillPascoe** [HughCraig/GHAP#260]
Now there are many layers, it can be hard to find them so they should be returned in a search. 
Add a tickbox to advanced search for 'Find Layers and Multilayers'. This applies the search to only the Layer and Multilayer metadata where the search fields are applicable (eg: title, description, options for NCC, ANPS, TLCMap etc etc).
Add a field to search 'keywords' also. This field only applies to Layer and Multilayers, so it should only be active if the tickbox to search Layers and Multilayers is ticked. This should be an autocomplete-as-the-user-types style input field, so the user can choose only from keywords that already exist, and can search multiple keywords at once (ie: it should work much like the keyword input field when creating a layer). 
Add these search options to the WS API GET parameters etc.

**Comment by IanMcCrabb** [comment]
@BillPascoe Would we not add search to Layers rather than Advanced Search?

**Comment by BillPascoe** [comment]
Good point. I feel like people would click 'Search' expecting all search functions there, and that they would be able to search there using 'keywords' and 'keywords' (Subject) are at the layer/ml level. However, these is really a distinct search with a distinct listing, so the 'search results' would need to be broken in two, not to mention the Advanced search screen would be very cluttered. So yes. Do it like this:

On the Layer page (and do same on Multilayer page), add a 'Find Layers' button, that opens a search form (this could be a concertina, or pop up window, a new page, whatever you think is best).

The search form enables search the following:

- Title and Description (textbox - do a 'contains' match)
- Type (multselect)
- Contributor  (autocomplete-as-you-type multiselect)
- Subject (autocomplete-as-you-type multiselect)
- Creator  (autocomplete-as-you-type multiselect)
- Publisher  (autocomplete-as-you-type multiselect)
- Citation (textbox - do a 'contains' match)
- DOI (textbox - do a 'contains' match)

Results should list the results just as they do on the layer page.

As always, if this doesn't seem right, and there is a better option, please recommend it.

**Comment by BillPascoe** [comment]
Revise this and recreate issues. We do need to include Layer 'keyword'/'subject' in the main advanced search screen. As Ian has suggested, the layer filtering should be on the layer page. This page already has a simple 'search' field, that filters down the list. This can be enhanced with advanced layer search, in order to eg: find layers with word match in title or description; find all layers with keyword 'history' etc.

**Comment by BillPascoe** [comment]
Also, see Autodetect Layer Geometries #47 and include geo search for layers.
